ruuvigw2mqtt/index.js

```diff
@@ -0,0 +1,399 @@
+const mqtt = require("mqtt");
+
+const ruuviParser = require("./ruuvi/parser");
+const dc = require("./device_config");
+const options = require(process.argv[2] || "/data/options.json");
+
+const timeBetweenData = options.update_interval * 1000; // ms
+const lastSent = {};
+const mqttOpts = {
+  protocol: options.mqtt_protocol,
+  host: options.mqtt_host,
+  port: options.mqtt_port,
+  clientId:
+    options.mqtt_credentials.client_id ||
+    "ruuvi-gw-mqtt-to-ha_" + Math.random().toString(16).substr(2, 8),
+  username: options.mqtt_credentials.username,
+  password: options.mqtt_credentials.password,
+};
+
+const client = mqtt.connect(mqttOpts);
+
+client.on("connect", () => {
+  console.log("mqtt connected");
+  client.subscribe(options.mqtt_ruuvi_topic);
+});
+
+client.on("close", () => {
+  console.log("mqtt disconnected");
+});
+
+client.on("error", (error) => {
+  console.error(`mqtt error ${error}`);
+});
+
+client.on("message", (topic, message) => {
+  const payload = JSON.parse(message.toString());
+  if (payload?.data) {
+    const mac = /[^/]*$/.exec(topic)[0];
+    handleData(payload, mac);
+  } else {
+    console.log(topic, message.toString());
+  }
+});
+
+const send = (id, payload) => {
+  console.log(`${options.mqtt_ha_topic}${id}`, JSON.stringify(payload));
+  client.publish(`${options.mqtt_ha_topic}${id}`, JSON.stringify(payload));
+};
+
+const sendConfig = (id, type, payload) => {
+  console.log(
+    `${options.mqtt_ha_topic}sensor/${id}/${type}/config`,
+    JSON.stringify(payload)
+  );
+
+  client.publish(
+    `${options.mqtt_ha_topic}sensor/${id}/${type}/config`,
+    JSON.stringify(payload),
+    { retain: true }
+  );
+};
+
+const handleData = (sr, mac) => {
+  mac = mac.split(":").join("");
+
+  // if (sr.data.indexOf("FF9904") !== -1) {
+  if (sr.data.startsWith("FF9904")) {
```

Comment:
This does not seem correct, 9904 is the companyID so it's not at the start of the package. https://docs.ruuvi.com/communication/bluetooth-advertisements

 Also, keep an eye out for the xiaomi sensors startsWith. Those strings is just what I found they seem to match 😅. The original code was from BT so there was localName to use for ID'ing sensor types.

Reply:
Ah, okay! Good that you noticed it 😅  I thought it will be at the start of the string, but feel free to fix these/those that I broke 😅

Reply:
I reverted these, as the RuuviTag parsing didn't work at all with the `startsWith()` 😅